node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -91,8 +94,73 @@ pub struct FailedTx {
     pub status: FailureStatus,
 }
 
+impl Transaction for FailedTx {
+    fn hash(&self) -> TxHash {
+        todo!()
+    }
+
+    fn receiver_address(&self) -> Address {
+        self.receiver_address
+    }
+
+    fn amount(&self) -> u128 {
+        todo!()
+    }
+
+    fn timestamp(&self) -> i64 {
+        todo!()
+    }
+
+    fn gas_price_wei(&self) -> u128 {
+        todo!()
+    }
+
+    fn nonce(&self) -> u64 {
+        todo!()
+    }
+
+    fn is_failed(&self) -> bool {
+        todo!()
+    }
+}
+
+// PartialOrd and Ord are used to create BTreeSet
+impl PartialOrd for FailedTx {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for FailedTx {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Descending Order
+        other
+            .timestamp
+            .cmp(&self.timestamp)
+            .then_with(|| other.nonce.cmp(&self.nonce))
+            .then_with(|| other.amount.cmp(&self.amount))
+    }
+}
+
+impl FailedTx {
+    pub fn from_sent_tx_and_web3_err(sent_tx: &Tx, error: &Web3Error) -> Self {
```

Comment:
This hurts my eyes. If you want to write idiomatic Rust then this is not the way. Only the `From<T>` is. 

You have two options. I like both. 

a) From<(&TX, &Web3Error)>

struct Tuple <'a> {
    sentx_tx: &'a Tx,
    error: &'a Web3Error
}

b) From<Tuple<'a>>

but a) just wins clearly 

Please accept that this is as normal in Rust as we say hello meeting somebody.